src/gsnClient/gsnTxHelpers.ts

```diff
@@ -242,7 +292,16 @@ export const getExecuteMetatransactionTx = async (
 ) => {
   const token = erc20(provider, contractAddress);
   const name = await token.name();
-  const nonce = await token.getNonce(account.address);
+  let nonce;
```

Comment:
Since we have this same code in 2 places, probably makes sense to extract to a method/function that both places can just call. Then you also wouldn't need the inline code comments since the flow would be really clear.

Reply:
updated